src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs

```diff
@@ -56,7 +56,7 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             Document document, ImmutableArray<Diagnostic> diagnostics,
             SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
         {
-            using var spansDisposer = ArrayBuilder<TextSpan>.GetInstance(diagnostics.Length, out var spans);
+            using var _ = ArrayBuilder<TextSpan>.GetInstance(diagnostics.Length, out var spans);
```

Comment:
no point naming these disposers.  our pattern in teh vast majority of other cases is just to use the `_` or `_1...` to signifiy this should not be accessed itself.

Reply:
Is there a point to _having_ the disposers, if the calls are changed to clear? I thought the whole idea was to save the need to call clear, but if you're not calling clear, should they just be removed?

Reply:
yes.  the disposers are necessary for two reasons:

1. ensuring we clean up in the event of early return (for example, cancellation).
2. ensuring the *builder* gets returned to the pool.  ToImmutableAndClear will give the resultant ImmutableArray and clear the builder.  but the builder is not returned to the pool.